File: src/main/java/com/placement/portal/service/PlacementDriveService.java
package com.placement.portal.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.placement.portal.dto.PlacementDriveRequestDTO;
import com.placement.portal.dto.PlacementDriveResponseDTO;

public interface PlacementDriveService {

    // ==========================================
    // Create Placement Drive
    // ==========================================

   PlacementDriveResponseDTO createPlacementDrive(

        PlacementDriveRequestDTO requestDTO,

        String recruiterEmail);;

    // ==========================================
    // Update Placement Drive
    // ==========================================

    PlacementDriveResponseDTO updatePlacementDrive(
            Long driveId,
            PlacementDriveRequestDTO requestDTO);

    // ==========================================
    // Get Placement Drive By ID
    // ==========================================

    PlacementDriveResponseDTO getPlacementDriveById(
            Long driveId);

    // ==========================================
    // Get All Placement Drives
    // ==========================================

    Page<PlacementDriveResponseDTO> getAllPlacementDrives(

            int page,

            int size,

            String sortBy,

            String direction);

    // ==========================================
    // Delete Placement Drive
    // ==========================================

    void deletePlacementDrive(Long driveId);
    List<PlacementDriveResponseDTO> getUpcomingPlacementDrives();
}

File: frontend/src/services/PlacementDriveService.js
import api from "../api/api";

export const getAllPlacementDrives = (page = 0) => {
    return api.get(`/placement-drives?page=${page}`);
};

export const getPlacementDriveById = (id) => {
    return api.get(`/placement-drives/${id}`);
};

export const getUpcomingDrives = () => {
    return api.get("/placement-drives/upcoming");
};

export const deletePlacementDrive = (id) => {
    return api.delete(`/placement-drives/${id}`);
};

export const createPlacementDrive = (data) => {
    return api.post("/placement-drives", data);
};

export const updatePlacementDrive = (id, drive) => {

    return api.put(
        `/placement-drives/${id}`,
        drive
    );

};


